Control/Controller.php

```diff
@@ -695,12 +695,15 @@ public function setSession(Session $session) {
 	 *
 	 * Caution: All parameters are expected to be URI-encoded already.
 	 *
-	 * @param string
-	 *
+	 * @param string|array $arg,.. One or more link segments, or list of link segments as an array
 	 * @return string
 	 */
-	public static function join_links() {
-		$args = func_get_args();
+	public static function join_links($arg = null) {
+		if (func_num_args() === 1 && is_array($arg)) {
```

Comment:
Because otherwise you can't pass in an arbitrary number of arguments; Each urlsegment needs to be passed explicitly as its own argument.